channeldb/channel.go

```diff
@@ -671,6 +688,81 @@ func (c *OpenChannel) MarkAsOpen(openLoc lnwire.ShortChannelID) error {
 	return nil
 }
 
+// MarkDataLoss marks sets the channel status to LocalDataLoss and stores the
+// passed commitPoint for use to retrieve funds in case the remote force closes
+// the channel.
+func (c *OpenChannel) MarkDataLoss(commitPoint *btcec.PublicKey) error {
+	c.Lock()
+	defer c.Unlock()
+
+	if err := c.Db.Update(func(tx *bolt.Tx) error {
```

Comment:
would be great to use `putChanStatus` here to reduce code duplication, signature might be more like:
```golang
func (c *OpenChannel) putChanStatus(tx *bolt.Tx, status ChannelStatus) (ChannelStatus, error)
```
for easier composition

Reply:
Tried doing this, but this turned out to now reduce duplication, since then the DB must be `View`ed from the caller, and the bucket must be retrieved twice in `MarkDataLoss`. The other suggestion were added, PTAL :)

Reply:
gotcha, thanks for trying anyway 😂